src/gsy_e/models/strategy/market_agents/two_sided_engine.py

```diff
@@ -149,8 +149,8 @@ def event_bid_traded(self, *, bid_trade):
 
             source_rate = bid_info.source_bid.energy_rate
             target_rate = bid_info.target_bid.energy_rate
-            assert abs(source_rate) + FLOATING_POINT_TOLERANCE >= abs(target_rate), \
-                f"bid: source_rate ({source_rate}) is not lower than target_rate ({target_rate})"
+            # assert abs(source_rate) + FLOATING_POINT_TOLERANCE >= abs(target_rate), \
+            #     f"bid: source_rate ({source_rate}) is not lower than target_rate ({target_rate})"
```

Comment:
In the case of DoF, someone could decide to buy energy from a certain seller at a higher price, however this assertion here is being raised due to that, so I was wondering whether it's still relevant or we should delete it

Reply:
There is a bug in the assertion message `lower` should be `higher` right?
Apart from that, I also do not remember why this assertion is here. IMO it is already a bit late to assert here. It should be in the propagation already.
If we decide to remove it, please also do the same for the one sided engine: https://github.com/gridsingularity/gsy-e/blob/989ee1edee8b104efca334d92518575fc8268448/src/gsy_e/models/strategy/market_agents/one_sided_engine.py#L155